File: source/kCura.Relativity.Export/Exporters/VolumeInfo.vb
Namespace kCura.Relativity.Export.Exports
	<Serializable()> Public Class VolumeInfo
		Implements System.Runtime.Serialization.ISerializable

#Region " Member and Accessors "

		Private _subdirectoryImagePrefix As String
		Private _subdirectoryNativePrefix As String
		Private _subdirectoryTextPrefix As String

		Public Property VolumePrefix() As String

		Public Property VolumeStartNumber() As Int32

		Public Property VolumeMaxSize() As Int64

		Public Property SubdirectoryStartNumber() As Int32

		Public Property SubdirectoryMaxSize() As Int64

		Public Property CopyFilesFromRepository() As Boolean = True

		Public Property SubdirectoryImagePrefix(Optional ByVal includeTopFolder As Boolean = True) As String
			Get
				Dim result As String = _subdirectoryImagePrefix
				If includeTopFolder Then
					result = "IMAGES\" + result
				End If
				Return result
			End Get
			Set(ByVal value As String)
				_subdirectoryImagePrefix = value
			End Set
		End Property

		Public Property SubdirectoryNativePrefix(Optional ByVal includeTopFolder As Boolean = True) As String
			Get
				Dim result As String = _subdirectoryNativePrefix
				If includeTopFolder Then
					result = "NATIVES\" + result
				End If
				Return result
			End Get
			Set(ByVal value As String)
				_subdirectoryNativePrefix = value
			End Set
		End Property

		Public Property SubdirectoryFullTextPrefix(Optional ByVal includeTopFolder As Boolean = True) As String
			Get
				Dim result As String = _subdirectoryTextPrefix
				If includeTopFolder Then
					result = "TEXT\" + result
				End If
				Return result
			End Get
			Set(ByVal value As String)
				_subdirectoryTextPrefix = value
			End Set
		End Property

#End Region

#Region " Constructors "
		Public Sub New()
			MyBase.New()
			Me.CopyFilesFromRepository = True
		End Sub
#End Region

#Region " Serialization "

		Public Sub GetObjectData(ByVal info As System.Runtime.Serialization.SerializationInfo, ByVal context As System.Runtime.Serialization.StreamingContext) Implements System.Runtime.Serialization.ISerializable.GetObjectData
			info.AddValue("CopyFilesFromRepository", Me.CopyFilesFromRepository, GetType(Boolean))
			info.AddValue("SubdirectoryMaxSize", Me.SubdirectoryMaxSize, GetType(Int64))
			info.AddValue("SubdirectoryStartNumber", Me.SubdirectoryStartNumber, GetType(Int32))
			info.AddValue("SubdirectoryFullTextPrefix", Me.SubdirectoryFullTextPrefix(False), GetType(String))
			info.AddValue("SubdirectoryNativePrefix", Me.SubdirectoryNativePrefix(False), GetType(String))
			info.AddValue("SubdirectoryImagePrefix", Me.SubdirectoryImagePrefix(False), GetType(String))
			info.AddValue("VolumeMaxSize", Me.VolumeMaxSize, GetType(Int64))
			info.AddValue("VolumeStartNumber", Me.VolumeStartNumber, GetType(Int32))
			info.AddValue("VolumePrefix", Me.VolumePrefix, GetType(String))
		End Sub
		Private Sub New(ByVal info As System.Runtime.Serialization.SerializationInfo, ByVal Context As System.Runtime.Serialization.StreamingContext)
			With info
				Me.CopyFilesFromRepository = .GetBoolean("CopyFilesFromRepository")
				Me.SubdirectoryMaxSize = .GetInt64("SubdirectoryMaxSize")
				Me.SubdirectoryStartNumber = .GetInt32("SubdirectoryStartNumber")
				Me.SubdirectoryFullTextPrefix = .GetString("SubdirectoryFullTextPrefix")
				Me.SubdirectoryNativePrefix = .GetString("SubdirectoryNativePrefix")
				Me.SubdirectoryImagePrefix = .GetString("SubdirectoryImagePrefix")
				Me.VolumeMaxSize = .GetInt64("VolumeMaxSize")
				Me.VolumeStartNumber = .GetInt32("VolumeStartNumber")
				Me.VolumePrefix = .GetString("VolumePrefix")
			End With
		End Sub
#End Region

	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Exporters/VolumeInfo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
namespace kCura.Relativity.Export.Exports
{
	[Serializable()]
	public class VolumeInfo : System.Runtime.Serialization.ISerializable
	{

		#region " Member and Accessors "

		private string _subdirectoryImagePrefix;
		private string _subdirectoryNativePrefix;

		private string _subdirectoryTextPrefix;
		public string VolumePrefix { get; set; }

		public Int32 VolumeStartNumber { get; set; }

		public Int64 VolumeMaxSize { get; set; }

		public Int32 SubdirectoryStartNumber { get; set; }

		public Int64 SubdirectoryMaxSize { get; set; }

		public bool CopyFilesFromRepository { get; set; }

		public string GetSubdirectoryImagePrefix(bool includeTopFolder) {
			
				string result = _subdirectoryImagePrefix;
				if (includeTopFolder) {
					result = "IMAGES\\" + result;
				}
				return result;
			
			
		}

	    public string SubdirectoryImagePrefix
	    {
	        get { return _subdirectoryImagePrefix; }
	        set { _subdirectoryImagePrefix = value; }
        }


        public string GetSubdirectoryNativePrefix(bool includeTopFolder)
        {
          
                string result = _subdirectoryNativePrefix;
                if (includeTopFolder)
                {
                    result = "NATIVES\\" + result;
                }
                return result;
           
       
        }
        public string SubdirectoryNativePrefix {
            get { return _subdirectoryNativePrefix; }
            set { _subdirectoryNativePrefix = value; }
		}

        public string GetSubdirectoryFullTextPrefix(bool includeTopFolder)
        {
                string result = _subdirectoryTextPrefix;
                if (includeTopFolder)
                {
                    result = "TEXT\\" + result;
                }
                return result;
        }

        public string SubdirectoryFullTextPrefix {
            get { return _subdirectoryTextPrefix; }
            set { _subdirectoryTextPrefix = value; }
		}

		#endregion

		#region " Constructors "
		public VolumeInfo() : base()
		{
			this.CopyFilesFromRepository = true;
		}
		#endregion

		#region " Serialization "

		public void GetObjectData(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context)
		{
			info.AddValue("CopyFilesFromRepository", this.CopyFilesFromRepository, typeof(bool));
			info.AddValue("SubdirectoryMaxSize", this.SubdirectoryMaxSize, typeof(Int64));
			info.AddValue("SubdirectoryStartNumber", this.SubdirectoryStartNumber, typeof(Int32));
			info.AddValue("SubdirectoryFullTextPrefix", this.GetSubdirectoryFullTextPrefix(false), typeof(string));
			info.AddValue("SubdirectoryNativePrefix", this.GetSubdirectoryNativePrefix(false), typeof(string));
			info.AddValue("SubdirectoryImagePrefix", this.GetSubdirectoryImagePrefix(false), typeof(string));
			info.AddValue("VolumeMaxSize", this.VolumeMaxSize, typeof(Int64));
			info.AddValue("VolumeStartNumber", this.VolumeStartNumber, typeof(Int32));
			info.AddValue("VolumePrefix", this.VolumePrefix, typeof(string));
		}
		private VolumeInfo(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext Context)
		{
			var _with1 = info;
			this.CopyFilesFromRepository = _with1.GetBoolean("CopyFilesFromRepository");
			this.SubdirectoryMaxSize = _with1.GetInt64("SubdirectoryMaxSize");
			this.SubdirectoryStartNumber = _with1.GetInt32("SubdirectoryStartNumber");
			this.SubdirectoryFullTextPrefix = _with1.GetString("SubdirectoryFullTextPrefix");
			this.SubdirectoryNativePrefix = _with1.GetString("SubdirectoryNativePrefix");
			this.SubdirectoryImagePrefix = _with1.GetString("SubdirectoryImagePrefix");
			this.VolumeMaxSize = _with1.GetInt64("VolumeMaxSize");
			this.VolumeStartNumber = _with1.GetInt32("VolumeStartNumber");
			this.VolumePrefix = _with1.GetString("VolumePrefix");
		}
		#endregion

	}
}
